lib/env.ts

```diff
@@ -0,0 +1,43 @@
+import { Option, fromNullable } from "fp-ts/lib/Option";
+import * as t from "io-ts";
+
+export const EnvMap = t.exact(
+  t.partial({
+    SPID_SESSION_TOKEN: t.string,
+    IO_BACKEND_HOST: t.string,
+    IO_BACKEND_BASEPATH: t.string
+  })
+);
+export type EnvMap = t.TypeOf<typeof EnvMap>;
+
+const pickEnvKeys = (obj: any) =>
+  Object.entries(obj).reduce(
+    (p, [k, v]) =>
+      EnvMap.decode({ [k]: v }).fold(
+        () => p,
+        (o: EnvMap) => ({ ...p, ...o })
+      ),
+    {}
+  );
+
+const env: EnvMap = { ...pickEnvKeys(process.env) };
```

Comment:
it's very complex to make a code review of a PR that has dangling code. can you please do some clean up ?